### ISSUE TITLE
Change CTRL+TAB shortcut to cycle through recent tabs

### ISSUE THREAD

**Issue by lewisc64** [godotengine/godot-proposals#8340]
### Describe the project you are working on

N/A - General use of the editor.

### Describe the problem or limitation you are having in your project

Control + Tab currently cycles through the tabs in order of left to right. This is consistent with the design of web browsers and windows explorer, but not with IDE's, text editors, and other coding environments (in my experience).

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Changing CTRL+TAB to cycle through the most recent tabs accessed would be a more expected (and more useful) behavior. Examples of software that does this:

- Notepad++
- Visual Studio
- Visual Studio Code
- Sublime text

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

The idea is in line with what notepad++ does.

When holding CTRL and pressing TAB, a box appears with list of tab names ordered with most recently accessed at the top.

On first press, the second item will be selected, because the top item is the currently selected tab. From here the user could do two things:

1. Release the control key

This would have the effect of switching to the tab you were previously on before the current one.

2. Still holding control, press and release the tab key.

This would set the highlighted tab to the next item down. Releasing control would close the box and select the tab in the editor.

3. Still holding control, use the mouse to click a tab in the history.

Upon clicking a tab in the box, the box would close and the editor would switch to that tab.

Here is a crude edited screenshot of the sort of thing I am talking about:
![image](https://github.com/godotengine/godot-proposals/assets/10404258/26591a64-0073-4585-97e9-4b89da4ae21c)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No

### Is there a reason why this should be core and not an add-on in the asset library?

I believe it should be the expected default behavior of this shortcut.

**Comment by AThousandShips** [comment]
Should be a *separate* shortcut IMO, as both navigation methods are very relevant (and it's not universal in software, Kate for example does navigate like Godot with these keys)

**Comment by cdemirer** [comment]
I would like this as well, however it might be better as an option, like in Firefox:

<img width="789" height="124" alt="Image" src="https://github.com/user-attachments/assets/ecde32c9-393d-4c8e-9a39-2041c2636fe1" />

Btw it looks like Kate has since changed to also use the recent order. (unless I simply can't find the setting that I may have previously enabled)

**Comment by geekley** [comment]
> Control + Tab currently cycles through the tabs in order of left to right. This is consistent with the design of web browsers

No, it's not. It's literally ONLY Google Chrome that refused to implement such an essential feature, asked since ever. Every decent browser (i.e. every other browser, including Firefox and every Chromium fork like Brave, Opera, Vivaldi) ends up going to the trouble of implementing at least the option to cycle using MRU (Most Recently Used) order because they refuse to do it upstream. It's unbelievable that there's still apps that don't implement this correctly, and it's all Chrome's fault. It goes completely against every IDE and the OS alt+tab and winkey+tab (Windows, Linux, ...) convention.

It's not just a mere convention, it's a productivity **necessity**.
Way too often you're working between two/three pages amongst many others (which are far apart) and you need to switch back and forth. This has always annoyed me on Godot. The last used tab is the tab you want to switch back to in 95% of cases; on the exceptions, you just click the tab you want. Without this feature, tab management is a pain because I have to close tabs all the time and have only the couple that I'm working on for this out-of-convention behavior to annoy me less...

> Should be a separate shortcut IMO, as both navigation methods are very relevant

@AThousandShips Having proper Ctrl+Tab doesn't mean losing the option to cycle using linear order -- it's important that you're still able to do that using Ctrl+PageDown (go right) and Ctrl+PageUp (go left). That's the established convention, just look at VSCode, Firefox, etc. Ctrl+Tab needs to use MRU order by default so Ctrl+Tab and Ctrl+Shift+Tab will cycle to the tab you actually want to go back to. I dunno if macOS/Safari uses a different convention, but in the Windows and Linux world that's the norm.

So this is how it should work by default:
- <kbd>Ctrl+Tab</kbd>: Go back in tab history. It's important to make this the default.
    - Users can easily just edit the shortcut keys to get the old behavior back if they want.
- <kbd>Ctrl+Shift+Tab</kbd>: Go forward in tab history (reverse of the above).
- <kbd>Ctrl+PageDown</kbd>: Go to the tab on the right (though for RTL languages I dunno what the convention is).
- <kbd>Ctrl+PageUp</kbd>: Go to the tab on the left (reverse of the above).
- Implement this for both scene switching and script switching. Hopefully in a way I can even set the same keys to switch either, depending on whether Godot is in Script mode or not.
- Please also implement the tab "history" list while holding Ctrl, as already described.
    - Preferably show it in the center of the screen.
    - While you're cycling, the scene/script you're going to is already shown even before you release Ctrl.
    - Then also make it so pressing <kbd>Esc</kbd> while this list is shown (i.e. you're still holding <kdb>Ctrl</kdb>) can cancel the switching and return to the same tab, like in Firefox.
    - You could add a setting to disable this list's visibility and/or set the delay (if any) until it's shown, while holding Ctrl. Only for tab history list visibility; it doesn't change behavior.  
    E.g. -1 = no list, 0 = show list immediately, default = 200ms delay.

> Is there a reason why this should be core and not an add-on in the asset library?

There's apparently [this addon](https://godotassetlibrary.com/asset/dLs8kP/previous-tab) in the asset lib, but I haven't tested it. In any case, it's an editor-only feature and you shouldn't have to put an addon in the project/VCS for this. It's expected functionality for any multi-tab app.
